fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -364,14 +305,8 @@ public <M extends Message> CompletableFuture<Void> update(@Nullable FDBIndexable
         return AsyncUtil.DONE;
     }
 
-    /**
-     * Get the {@link AnalyzingInfixSuggester} for indexing or query, from the session of the context if there exists a corresponding one, or by creating a new one.
-     * @param indexOptions the {@link IndexOptions} for suggester's {@link FieldType}. This only matters for when the suggester is for indexing.
-     * The one for query can just use an arbitrary one, so just pass in a NULL when getting a suggester for query, so the existing one from session of context can be reused.
-     */
-    private AnalyzingInfixSuggester getSuggester(@Nullable Tuple groupingKey, @Nonnull List<String> texts, @Nullable IndexOptions indexOptions) throws IOException {
-        return directoryManager.getAutocompleteSuggester(groupingKey, autoCompleteIndexAnalyzerChooser.chooseAnalyzer(texts),
-                autoCompleteQueryAnalyzerChooser.chooseAnalyzer(texts), highlightForAutoCompleteIfEnabled, indexOptions);
+    private Analyzer getAutocompleteQueryAnalyzer(@Nonnull List<String> texts) {
+        return autoCompleteQueryAnalyzerChooser.chooseAnalyzer(texts).getAnalyzer();
```

Comment:
Considering the auto-complete uses the same index as basic index now, not sure whether it still needs a separate query analyzer. I think this depends on the behavior we want to support and we can further discuss this offline.